tests/robustness/scenarios.go

```diff
@@ -69,22 +69,19 @@ func exploratoryScenarios(_ *testing.T) []testScenario {
 			options.ClusterOptions{options.WithTickMs(100), options.WithElectionMs(2000)}),
 	}
 
-	mixedVersionOption := options.WithClusterOptionGroups(
-		// 60% with all members of current version
-		options.ClusterOptions{options.WithVersion(e2e.CurrentVersion)},
-		options.ClusterOptions{options.WithVersion(e2e.CurrentVersion)},
-		options.ClusterOptions{options.WithVersion(e2e.CurrentVersion)},
-		options.ClusterOptions{options.WithVersion(e2e.CurrentVersion)},
-		options.ClusterOptions{options.WithVersion(e2e.CurrentVersion)},
-		options.ClusterOptions{options.WithVersion(e2e.CurrentVersion)},
-		// 10% with 2 members of current version, 1 member last version, leader is current version
-		options.ClusterOptions{options.WithVersion(e2e.MinorityLastVersion), options.WithInitialLeaderIndex(0)},
-		// 10% with 2 members of current version, 1 member last version, leader is last version
-		options.ClusterOptions{options.WithVersion(e2e.MinorityLastVersion), options.WithInitialLeaderIndex(2)},
-		// 10% with 2 members of last version, 1 member current version, leader is last version
-		options.ClusterOptions{options.WithVersion(e2e.QuorumLastVersion), options.WithInitialLeaderIndex(0)},
-		// 10% with 2 members of last version, 1 member current version, leader is current version
-		options.ClusterOptions{options.WithVersion(e2e.QuorumLastVersion), options.WithInitialLeaderIndex(2)},
+	mixedVersionOption := options.WithWeightedClusterOptionGroups(
+		[]options.ClusterOptions{
+			// 60% with all members of current version
+			{options.WithVersion(e2e.CurrentVersion)},
+			// 10% with 2 members of current version, 1 member last version, leader is current version
+			{options.WithVersion(e2e.MinorityLastVersion), options.WithInitialLeaderIndex(0)},
+			// 10% with 2 members of current version, 1 member last version, leader is last version
+			{options.WithVersion(e2e.MinorityLastVersion), options.WithInitialLeaderIndex(2)},
+			// 10% with 2 members of last version, 1 member current version, leader is last version
+			{options.WithVersion(e2e.QuorumLastVersion), options.WithInitialLeaderIndex(0)},
+			// 10% with 2 members of last version, 1 member current version, leader is current version
+			{options.WithVersion(e2e.QuorumLastVersion), options.WithInitialLeaderIndex(2)}},
+		[]float64{0.6, 0.1, 0.1, 0.1, 0.1},
```

Comment:
That's less readable than previous solution, by weight options like traffic, I literally meant the weighted choice from robustness traffic. 

https://github.com/etcd-io/etcd/blob/7a3ca533456f838fc9cc4c60997dbea303825f2d/tests/robustness/traffic/random.go#L31-L49

Example of usage:

https://github.com/etcd-io/etcd/blob/7a3ca533456f838fc9cc4c60997dbea303825f2d/tests/robustness/traffic/etcd.go#L36-L48

Reply:
done